### PR TITLE
[TypeDeclaration] Test native-type ArrayDimFetch on numeric keys

### PR DESCRIPTION
**Pull request by staabm**
more tests to cover https://github.com/rectorphp/rector-src/pull/5056

### CODE REVIEW

File: rules-tests/TypeDeclaration/Rector/ClassMethod/AddMethodCallBasedStrictParamTypeRector/Fixture/native_array_shape3.php.inc [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+-----
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\ClassMethod\AddMethodCallBasedStrictParamTypeRector\Fixture;
+
+final class NativeArrayShape3 {
+    private function doFoo() {
+        $arr = [];
+        $arr[10] = [123.132];
+        $this->doBar($arr[10]);
```
Comment (samsonasik) [inline rules-tests/TypeDeclaration/Rector/ClassMethod/AddMethodCallBasedStrictParamTypeRector/Fixture/native_array_shape3.php.inc:26]: The fixture is copy paste incorrect, the key is `0` :)

```suggestion
        $arr[0] = [123.132];
        $this->doBar($arr[0]);
```

Comment (staabm) [reply rules-tests/TypeDeclaration/Rector/ClassMethod/AddMethodCallBasedStrictParamTypeRector/Fixture/native_array_shape3.php.inc:26]: args, right.. ;)

**Comment by samsonasik** [comment]
It seems working ok, I am merging it ;)

**Comment by staabm** [comment]
Yeah, I was suprised tbh

**Pull request merged**